### PR TITLE
Reference tests for display:contents

### PR DESCRIPTION
**Pull request by rune-opera**
It was not clear to me from the spec what the expected result of applying display:contents to replaced elements is. The added test for replace contents matches what Gecko implemented.

---
This change is [Reviewable](https://reviewable.io/reviews/w3c/csswg-test/1129)

### CODE REVIEW

File: css-display-3/display-contents-first-line-001.html [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>CSS Display: display:contents and ::first-line</title>
```
Comment (zcorpan) [inline css-display-3/display-contents-first-line-001.html]: (This test seems to fail in Gecko, though that seems like a bug in Gecko. Filed https://bugzilla.mozilla.org/show_bug.cgi?id=1305951 )

File: css-display-3/display-contents-td-001.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>CSS Display: Children of multiple td elements with display:contents wrapped in single anonymous table cell</title>
+<link rel="author" title="Rune Lillesveen" href="mailto:rune@opera.com">
+<link rel="help" href="https://drafts.csswg.org/css-display-3/#valdef-display-contents">
+<link rel="match" href="display-contents-td-001-ref.html">
+<style>
+    td { display: contents }
```
Comment (zcorpan) [inline css-display-3/display-contents-td-001.html]: Maybe add more padding so it's very clear when it fails

File: css-display-3/display-contents-td-001-ref.html [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>CSS Reftest Reference</title>
+<link rel="author" title="Rune Lillesveen" href="mailto:rune@opera.com">
+<p>You should see the word PASS without extra spacing below.</p>
+<table cellpadding="0">
+    <td>PASS</td>
```
Comment (rune-opera) [inline css-display-3/display-contents-td-001-ref.html]: Done

File: css-display-3/display-contents-td-001.html (hunk 2 above)
Comment (rune-opera) [inline css-display-3/display-contents-td-001.html]: Done

File: css-display-3/display-contents-td-001.html [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+</style>
+<p>You should see the word PASS without extra spacing below.</p>
+<table>
+    <td>P</td><td>A</td><td>S</td><td>S</td>
```
Comment (rune-opera) [inline css-display-3/display-contents-td-001.html]: Done

File: css-display-3/display-contents-tr-001-ref.html [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>CSS Reftest Reference</title>
+<link rel="author" title="Rune Lillesveen" href="mailto:rune@opera.com">
+<p>You should see the word PASS below.</p>
+<table cellpadding="0" cellspacing="0">
+    <td>PASS</td>
```
Comment (rune-opera) [inline css-display-3/display-contents-tr-001-ref.html]: Done

File: css-display-3/display-contents-first-line-001.html (hunk 1 above)
Comment (rune-opera) [inline css-display-3/display-contents-first-line-001.html]: Ack

File: css-display-3/display-contents-before-after-001-ref.html [hunk 6]
```diff
@@ -0,0 +1,6 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>CSS Reftest Reference</title>
+<link rel="author" title="Rune Lillesveen" href="mailto:rune@opera.com">
+<p>You should see the word PASS below.</p>
+PASS
```
Comment (rune-opera) [inline css-display-3/display-contents-before-after-001-ref.html]: I've re-organized a bit to do this. I split the margin-collapsing through display:contents into a separate test (block-002)

**Pull request merged**